### PR TITLE
task4593 post list

### PR DESCRIPTION
**Pull request by laam55**
https://edu-redmine.framgia.vn/issues/4593

### CODE REVIEW

File: app/Repositories/Eloquents/PostRepository.php [hunk 1]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
     {
         return $this->model()->destroy($id);
     }
+
+    // post index
+    public function getPostWithUserLimit($input, $limit = null)
+    {
+        $limit = (is_null($limit)) ? config('constants.PAGINATION_LIMIT_NUMBER',  16) : $limit;
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.TooMuchSpaceAfterComma</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 91</br>Expected 1 space after comma in function call; 2 found

File: app/Repositories/Eloquents/PostRepository.php [hunk 2]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+        $query = $this->model()->with('user')->with('tags');
+
+        return $this->search($query, $input, $limit);
+    }
+
+    // get post with paginate at post all, search module
+    public function getPostWithUserPaginate($input, $limit = null)
+    {
+        $limit = (is_null($limit)) ? config('constants.PAGINATION_LIMIT_NUMBER',  16) : $limit;
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.TooMuchSpaceAfterComma</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 100</br>Expected 1 space after comma in function call; 2 found

File: app/Repositories/Eloquents/PostRepository.php [hunk 3]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+        $query = $this->model()->with('user')->with('tags');
+
+        return $this->search($query, $input, $limit, true);
+    }
+
+    public function getTreadingPostLimit($limit = null)
+    {
+        $limit = (is_null($limit)) ? config('constants.PAGINATION_LIMIT_TREDINGPOST',  16) : $limit;
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.TooMuchSpaceAfterComma</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 108</br>Expected 1 space after comma in function call; 2 found

File: app/Repositories/Eloquents/PostRepository.php [hunk 4]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+        $query = $this->model()->with('tags')->orderBy('total_view', 'desc')->orderBy('id', 'desc')->take($limit)->get();
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 109</br>Line exceeds 120 characters; contains 121 characters

File: app/Repositories/Eloquents/PostRepository.php [hunk 5]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+
+        return $query;
+    }
+
+    public function getOwnTags()
+    {
+        $limit = config('constants.PAGINATION_LIMIT_OWN_TAG',  16);
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.TooMuchSpaceAfterComma</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 116</br>Expected 1 space after comma in function call; 2 found

File: app/Repositories/Eloquents/PostRepository.php [hunk 6]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+
+        foreach ($tags as $key => $value) {
+            $ownTags = $ownTags->merge($value);
+        }
+
+        return $ownTags->unique('id')->take($limit);
+    }
+
+    // search treding, week, month post
+    public function search($query, $input, $limit, $isPaginate = false) {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 129</br>Opening brace should be on a new line

File: app/Repositories/Eloquents/PostRepository.php [hunk 7]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+        $appends = [];
+        $timeControl = new \Carbon\Carbon();
+
+        $startWeek = $timeControl->startOfWeek()->toDateString();
+        $endWeek = $timeControl->endOfWeek()->toDateString();
+        $startMonth = $timeControl->startOfMonth()->toDateString();
+        $endMonth = $timeControl->endOfMonth()->toDateString();
+
+        // filter by tab
+        if (!empty($input['tab'])) {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 139</br>Blank line found at start of control structure

File: app/Repositories/Eloquents/PostRepository.php [hunk 8]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+
+            if ($input['tab'] == 'treding') {
+                $query->orderBy('total_view', 'desc');
+
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingBeforeClose</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 143</br>Blank line found at end of control structure

File: app/Repositories/Eloquents/PostRepository.php [hunk 9]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+            } elseif ($input['tab'] == 'week') {
+                $query->whereBetween('created_at', array($startWeek, $endWeek));
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.ShortArraySyntax</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 145</br>"array()" function used. Use short array syntax, "[]", instead.

File: app/Repositories/Eloquents/PostRepository.php [hunk 10]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingBeforeClose</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 146</br>Blank line found at end of control structure

File: app/Repositories/Eloquents/PostRepository.php [hunk 11]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+            } elseif ($input['tab'] == 'month') {
+                $query->whereBetween('created_at', array($startMonth, $endMonth));
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.ShortArraySyntax</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 148</br>"array()" function used. Use short array syntax, "[]", instead.

File: app/Repositories/Eloquents/PostRepository.php [hunk 12]
```diff
@@ -84,4 +84,79 @@ public function delete($id)
...
+            }
+        } else {
+            $query->orderBy('total_vote', 'desc');
+        }
+
+        if (!empty($input['q'])) {
+            $query->where('title', 'like', '%' . $input['q'] . '%');
+        }
+        // is paginate?
+        $query = $isPaginate == true ? $query->orderBy('id', 'desc')->paginate($limit)->appends($appends) : $query->orderBy('id', 'desc')->take($limit)->get();
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 158</br>Line exceeds 120 characters; contains 159 characters

File: resources/views/home/post/index.blade.php [hunk 13]
```diff
@@ -1,35 +1,41 @@
...
             {{ __('page.post.list') }}
         </h1>
-        <div class="pl8 aside-cta grid--cell" role="navigation" aria-label="ask new question">
+        <div class="pl8 aside-cta grid--cell">
             <a href="#" class="d-inline-flex ai-center ws-nowrap s-btn s-btn__primary">
                 {{ __('page.post.write') }}
             </a>
         </div>
     </div>
+    <?php $input['tab'] = !empty($input['tab']) ? $input['tab'] : ''; ?>
```
Comment (BunnyPi04) [inline resources/views/home/post/index.blade.php]: @php nhé

File: resources/views/home/post/index.blade.php [hunk 14]
```diff
@@ -1,35 +1,41 @@
...
             <div class="grid tabs-filter s-btn-group tt-capitalize">
-                <a href="#" class="youarehere is-selected grid--cell s-btn s-btn__muted s-btn__outlined py8 ws-nowrap" data-title="Questions that may be of interest to you based on your history and tag preference">
+                <a href="{{ route('home.post.index') }}"
+                    class="{{ empty($input['tab']) ? 'is-selected' : '' }} youarehere grid--cell s-btn s-btn__muted s-btn__outlined py8 ws-nowrap">
                     {{ __('page.post.interesting') }}
                 </a>
-                <a href="#" class="grid--cell s-btn s-btn__muted s-btn__outlined py8 ws-nowrap" data-title="Questions with an active bounty">
-                    <span class="bounty-indicator-tab">363</span>{{ __('page.post.treding') }}
+                <a href="{{ route('home.post.index', ['tab' => 'treding']) }}"
+                    class="{{ ($input['tab'] == 'treding') ? 'is-selected' : '' }} grid--cell s-btn s-btn__muted s-btn__outlined py8 ws-nowrap">
```
Comment (BunnyPi04) [inline resources/views/home/post/index.blade.php]: trending not treding
sửa các chỗ tương tự

File: resources/views/home/post/index.blade.php [hunk 15]
```diff
@@ -38,87 +44,122 @@
...
+                                    <div>{{ trans_choice('page.post.answers', $post->total_answer) }}</div>
+                                </div>
+                            @else
+                                <div class="status {{ ($post->total_answer > 0) ? 'answered' : '' }}">
+                                    <div class="mini-counts"><span>{{ $post->total_answer }}</span></div>
+                                    <div>{{ trans_choice('page.post.answers', $post->total_answer) }}</div>
+                                </div>
+                            @endif
+                            <div class="views">
+                                <div class="mini-counts {{ ($post->total_view > 100) ? 'hot' : '' }}"><span>{{ $post->total_view }}</span></div>
```
Comment (BunnyPi04) [inline resources/views/home/post/index.blade.php]: số 100 này phải đưa config chứ?

**Pull request merged**